doc/api/test.md

```diff
@@ -1044,6 +1044,134 @@ exports[`suite of snapshot tests > snapshot test 2`] = `
 Once the snapshot file is created, run the tests again without the
 `--test-update-snapshots` flag. The tests should pass now.
 
+### Inline snapshots
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1.0 - Early development
+
+Inline snapshots store the expected snapshot content directly within the test
+file as a template literal, rather than in an external `.snapshot` file. This
+makes it easier to see what the test expects and reduces the number of files in
+your project. Inline snapshots are particularly useful for small snapshots
+where having the expected value close to the test provides better readability.
+
+The `t.assert.inlineSnapshot()` function creates inline snapshot assertions.
+When called without a second argument, the function will insert the serialized
+value as a template literal directly into your source code when run with the
+[`--test-update-snapshots`][] flag.
+
+```mjs
+import { test } from 'node:test';
+
+test('inline snapshot test', (t) => {
+  const user = { name: 'Alice', age: 30 };
+
+  // Initially call without second argument
+  t.assert.inlineSnapshot(user);
+  t.assert.inlineSnapshot('hello world');
+});
+```
+
+```cjs
+const { test } = require('node:test');
+
+test('inline snapshot test', (t) => {
+  const user = { name: 'Alice', age: 30 };
+
+  // Initially call without second argument
+  t.assert.inlineSnapshot(user);
+  t.assert.inlineSnapshot('hello world');
+});
+```
+
+When first run without the `--test-update-snapshots` flag, the test will fail
+with a "Missing inline snapshots" error. Run the test with
+`--test-update-snapshots` to generate the snapshots:
+
+```bash
+node --test --test-update-snapshots test.js
+```
+
+After running with the update flag, your test file will be automatically
+modified to include the snapshot content:
+
+```mjs
+import { test } from 'node:test';
+
+test('inline snapshot test', (t) => {
+  const user = { name: 'Alice', age: 30 };
+
+  t.assert.inlineSnapshot(user, `
```

Comment:
It would be really helpful if inlineSnapshot() supported wrapping. For example:

```ts
/**
 * Asserts bitmap is the expected hex value. The `expectedHex` arg is replaced
 * automatically when tests are run with `--test-update-snapshots`.
 */
function assertBitmapHex(ctx: TestContext, bmp: Uint8Array, expectedHex: string): string {
  const actualHex = [...bmp]
    .map(v => v.toString(16).padStart(2, '0'))
    .join('')
  ctx.assert.inlineSnapshot(actualHex, expectedHex) // <-- don't update me inline here!
}
```

```ts
assertBitmapHex(bmp, '000000000000000000000000')
```

It's possible to not use wrapping but then it gets pretty clumsy trying to fill out all the args in every test.